src/backend/distributed/transaction/transaction_management.c

```diff
@@ -185,6 +193,64 @@ InCoordinatedTransaction(void)
 }
 
 
+/*
+ * Sets a flag to true indicating that the current node is executing a delegated
+ * function call, using forcePushdown, within a distributed transaction issued
+ * by the coordinator. Also, saves the distribution argument.
+ */
+void
+EnableInForceDelegatedFuncExecution(Const *distArgument)
+{
+	MemoryContext oldcontext = MemoryContextSwitchTo(TopTransactionContext);
```

Comment:
could use a comment on why this needs to be in TopTransactionContext. Also wondering whether there is a less broad context.

Reply:
>>Also wondering whether there is a less broad context.
Likely there is which I'm missing, tried lot of contexts, and to my surprise MessageContext too didn't work. The reason being, execution happens in two phases, bind (where we save) and exec (where we compare)

Reply:
Added the comment